tools/interop_matrix/run_interop_matrix_tests.py

```diff
@@ -0,0 +1,190 @@
+    skipped = len(docker_image_list) - len(tags)
+    jobset.message('START', 'Skipped images (no-tag/unknown-tag): %d' % skipped,
+                   do_newline=True)
+    # Filter tags based on the releases.
+    images[runtime] = [(tag,'%s:%s' % (image_path,tag)) for tag in tags if
+                       tag in releases]
+  return images
+
+# caches test cases (list of JobSpec) loaded from file.  Keyed by lang and runtime.
+_loaded_testcases = {}
+def find_test_cases(lang, release):
+  """Returns the list of test cases from testcase files per lang/release."""
+  file_tmpl = os.path.join(os.path.dirname(__file__), 'testcases/%s__%s')
+  if not os.path.exists(file_tmpl % (lang, release)):
+    release = 'master'
+  testcases = file_tmpl % (lang, release)
+  if lang in _loaded_testcases.keys() and release in _loaded_testcases[lang].keys():
+    return _loaded_testcases[lang][release]
+
+  job_spec_list=[]
+  try:
+    with open(testcases) as f:
+      # Only line start with 'docker run' are test cases.
+      for line in f.readlines():
+        if line.startswith('docker run'):
+          line = line.replace('docker run', 'gcloud docker -- run')
+          m = re.search('--test_case=(.*)"', line)
+          shortname = m.group(1) if m else 'unknown_test'
+          spec = jobset.JobSpec(cmdline=line,
+                                shortname=shortname,
+                                timeout_seconds=_TEST_TIMEOUT,
+                                shell=True)
+          job_spec_list.append(spec)
+      jobset.message('START',
+                     'Loaded %s tests from %s' % (len(job_spec_list), testcases),
+                     do_newline=True)
+  except IOError as err:
+    jobset.message('FAILED', err, do_newline=True)
+  if lang not in _loaded_testcases.keys():
+    _loaded_testcases[lang] = {}
+  _loaded_testcases[lang][release]=job_spec_list
+  return job_spec_list
+
+_xml_report_tree = None
+def run_tests_for_lang(lang, runtime, images):
+  """Find and run all test cases for a language.
+
+  images is a list of (<release-tag>, <image-full-path>) tuple.
+  """
+  for image_tuple in images:
+    release, image = image_tuple
+    jobset.message('START', 'Testing %s' % image, do_newline=True)
+    _docker_images_cleanup.append(image)
+    job_spec_list = find_test_cases(lang,release)
+    num_failures, resultset = jobset.run(job_spec_list,
+                                         newline_on_success=True,
+                                         add_env={'docker_image':image},
+                                         maxjobs=args.jobs)
+    if num_failures:
+      jobset.message('FAILED', 'Some tests failed', do_newline=True)
+    else:
+      jobset.message('SUCCESS', 'All tests passed', do_newline=True)
+
+    # Required, otherwise _xml_report_tree will be shadowed by local (undefined)
+    # reference in the next line.
+    global _xml_report_tree
+    _xml_report_tree = report_utils.add_junit_xml_results(
```

Comment:
btw, both jenkins and kokoro can digest  multiple XML report files, so perhaps the functionality to merge XML reports from different languages is not useful?